LoopKitUI/Views/SizeablePicker.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  SizeablePicker.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 12/7/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+import UIKit
+
+struct SizeablePicker<SelectionValue>: UIViewRepresentable where SelectionValue: CustomStringConvertible & Hashable {
+    private let selection: Binding<SelectionValue>
+    private var selectedRow: Int = 0
+    // TODO: Would be nice if we could just use `ForEach` and Content, but for now, this'll do
+    private let data: [SelectionValue]
+    private let formatter: (SelectionValue) -> String
+    private let colorer: (SelectionValue) -> Color
+
+    public init(selection: Binding<SelectionValue>,
+                data: [SelectionValue],
+                formatter: @escaping (SelectionValue) -> String = { $0.description },
+                colorer: @escaping (SelectionValue) -> Color = { _ in .primary }
+    ) {
+        self.selection = selection
+        self.selectedRow = data.firstIndex(of: selection.wrappedValue) ?? 0
+        self.data = data
+        self.formatter = formatter
+        self.colorer = colorer
+    }
+
+    //makeCoordinator()
+    func makeCoordinator() -> SizeablePicker.Coordinator {
+        Coordinator(self)
+    }
+
+    //makeUIView(context:)
+    func makeUIView(context: UIViewRepresentableContext<SizeablePicker>) -> UIPickerView {
+        let picker = UIPickerViewResizeable(frame: .zero)
+        
+        picker.dataSource = context.coordinator
+        picker.delegate = context.coordinator
+
+        return picker
+    }
+
+    //updateUIView(_:context:)
+    func updateUIView(_ view: UIPickerView, context: UIViewRepresentableContext<SizeablePicker>) {
+        if view.selectedRow(inComponent: 0) != selectedRow {
+            view.selectRow(selectedRow, inComponent: 0, animated: true)
+        }
+    }
+
+    class Coordinator: NSObject, UIPickerViewDataSource, UIPickerViewDelegate {
+        var parent: SizeablePicker
```

Comment:
Nit: "Parent" is an kind of an odd term for this relationship. It's not a parent in a view or object hierarchy sense. Maybe `picker`, or `pickerView`?